pandas/tests/indexes/period/test_period.py

```diff
@@ -773,3 +773,9 @@ def test_map(self):
         result = index.map(lambda x: x.ordinal)
         exp = Index([x.ordinal for x in index])
         tm.assert_index_equal(result, exp)
+
+    @pytest.mark.parametrize('how', ['outer', 'inner', 'left', 'right'])
+    def test_join_self(self, how):
```

Comment:
Can you add this to tests/indexes/common.py using `self.create_index` ? The this should automatically be run for all index types